internal/controller/authenticator/jwtcachefiller/jwtcachefiller_test.go

```diff
@@ -363,41 +712,217 @@ func TestController(t *testing.T) {
 					Spec: *invalidTLSJWTAuthenticatorSpec,
 				},
 			},
-			wantErr: testutil.WantExactErrorString("failed to build jwt authenticator: invalid TLS configuration: illegal base64 data at input byte 7"),
+			wantStatusConditions: status.ReplaceConditions(
+				allHappyConditionsSuccess(someOtherIssuer, frozenMetav1Now, 0),
+				[]metav1.Condition{
+					sadReadyCondition(frozenMetav1Now, 0),
+					sadTLSConfigurationValid(frozenMetav1Now, 0),
+					unknownDiscoveryURLValid(frozenMetav1Now, 0),
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					unknownJWKSURLValid(frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase:  "Error",
+			wantCacheEntries: 0,
+		}, {
+			name: "validateIssuer: parsing error (spec.issuer URL is invalid) will fail sync loop and will report failed and unknown conditions and Error phase, but will not enqueue a resync due to user config error",
+			jwtAuthenticators: []runtime.Object{
+				&auth1alpha1.JWTAuthenticator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-name",
+					},
+					Spec: *invalidIssuerJWTAuthenticatorSpec,
+				},
+			},
+			syncKey: controllerlib.Key{Name: "test-name"},
+			wantStatusConditions: status.ReplaceConditions(
+				allHappyConditionsSuccess(goodIssuer, frozenMetav1Now, 0),
+				[]metav1.Condition{
+					sadReadyCondition(frozenMetav1Now, 0),
+					sadIssuerURLValidInvalid("https://.café   .com/café/café/café/coffee", frozenMetav1Now, 0),
+					unknownDiscoveryURLValid(frozenMetav1Now, 0),
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					unknownJWKSURLValid(frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase: "Error",
+		}, {
+			name: "validateIssuer: parsing error (spec.issuer URL has invalid scheme, requires https) will fail sync loop and will report failed and unknown conditions and Error phase, but will not enqueue a resync due to user config error",
+			jwtAuthenticators: []runtime.Object{
+				&auth1alpha1.JWTAuthenticator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-name",
+					},
+					Spec: *invalidIssuerSchemeJWTAuthenticatorSpec,
+				},
+			},
+			syncKey: controllerlib.Key{Name: "test-name"},
+			wantStatusConditions: status.ReplaceConditions(
+				allHappyConditionsSuccess(goodIssuer, frozenMetav1Now, 0),
+				[]metav1.Condition{
+					sadReadyCondition(frozenMetav1Now, 0),
+					sadIssuerURLValidInvalidScheme("http://.café.com/café/café/café/coffee", frozenMetav1Now, 0),
+					unknownDiscoveryURLValid(frozenMetav1Now, 0),
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					unknownJWKSURLValid(frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase: "Error",
+		}, {
+			name: "validateProviderDiscovery: could not perform oidc discovery on provider issuer will fail sync loop and will report failed and unknown conditions and Error phase and will enqueue new sync",
```

Comment:
This test fails on my Mac laptop.

Reply:
```
        	Error:      	Not equal: 
        	            	expected: "could not perform oidc discovery on provider issuer: Get \"https://127.0.0.1:443/some-other-issuer/.well-known/openid-configuration\": dial tcp 127.0.0.1:443: connect: connection refused"
        	            	actual  : "could not perform oidc discovery on provider issuer: Get \"https://127.0.0.1:443/some-other-issuer/.well-known/openid-configuration\": EOF"
```

I'm using Go 1.21.6 in Goland, in case that matters. Same failure for Go 1.22 also.

Reply:
Oh, this is because I really have a server running on port 443 on my Mac. The test could use its own fake OIDC provider server instead of assuming that port 443 is unused.

Reply:
Nice, used `goodIssuer` + `/foo/bar/baz` to make use of existing created server. 👍